test/gurka/test_sump_buster.cc

```diff
@@ -0,0 +1,151 @@
+  auto result = gurka::route(map, "A", "I", "auto");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  auto leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE", "EH", "HL", "KL", "JK", "IJ"});
+
+  result = gurka::route(map, "A", "I", "hov");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE", "EH", "HL", "KL", "JK", "IJ"});
+
+  result = gurka::route(map, "A", "I", "taxi");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DE", "EH", "HL", "KL", "JK", "IJ"});
+
+  result = gurka::route(map, "A", "I", "bus");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BF", "FI"});
+
+  result = gurka::route(map, "A", "I", "truck");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BF", "FI"});
+
+  result = gurka::route(map, "A", "I", "bicycle");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BF", "FI"});
+
+  result = gurka::route(map, "A", "I", "motor_scooter");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BF", "FI"});
+
+  result = gurka::route(map, "A", "I", "motorcycle");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BF", "FI"});
+
+  result = gurka::route(map, "A", "I", "pedestrian");
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  leg = result.trip().routes(0).legs(0);
+  gurka::assert::raw::expect_path(result, {"AB", "BF", "FI"});
+}
+
+TEST(Standalone, SumpBusterOverride) {
+  const std::string ascii_map = R"(
+      A----B----C----D------------E
+           |         |            |
+           F         G            H
+           |         |            |
+           I----J----K------------L
+  )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "residential"}}}, {"BC", {{"highway", "residential"}}},
+      {"CD", {{"highway", "residential"}}}, {"DE", {{"highway", "residential"}}},
+      {"BF", {{"highway", "residential"}}}, {"DG", {{"highway", "residential"}}},
+      {"EH", {{"highway", "residential"}}}, {"FI", {{"highway", "residential"}}},
+      {"GK", {{"highway", "residential"}}}, {"HL", {{"highway", "residential"}}},
+      {"IJ", {{"highway", "residential"}}}, {"JK", {{"highway", "residential"}}},
+      {"KL", {{"highway", "residential"}}},
+  };
+
+  const gurka::nodes nodes = {{"F",
+                               {{"barrier", "sump_buster"},
```

Comment:
This is more a question for my understanding: If a mode is not specified here (eg: `hov` or `motorcar`), does that mean this restriction does _not_ apply to it?

In the tests below, none of the modes use `F` (which has `motorcar: yes`). Does that mean `F` does NOT have the sump_buster restriction for anything except a `motorcar`?

Reply:
Also, this testcase can be shortened a quite a bit is parametrized fixtures are used (since the expected result is the same for all modes).

Reply:
@mandeepsandhu For sump_buster we allow access for bus, truck, motorcycle, moped, foot, and bicycle and block access for auto, hov, and taxi for the defaults.  However, if a user adds a mode specific key then that overrides the default.  This same logic is used for gates, bollards, and barriers.  This is how I found the access issue and why I added a test for it above.

Reply:
> Also, this testcase can be shortened a quite a bit is parametrized fixtures are used (since the expected result is the same for all modes).

I will update.

Reply:
> @mandeepsandhu For sump_buster we allow access for bus, truck, motorcycle, moped, foot, and bicycle and block access for auto, hov, and taxi for the defaults. However, if a user adds a mode specific key then that overrides the default. This same logic is used for gates, bollards, and barriers. This is how I found the access issue and why I added a test for it above.

Got it. Thanks for the explanation 🙏